### PR TITLE
Improve database configuration docs

### PR DESCRIPTION
**Pull request by richvdh**
Attempts to clarify the sample config for databases, and add some stuff about
tcp keepalives to `postgres.md`.

### CODE REVIEW

File: synapse/config/_base.py [hunk 1]
```diff
@@ -294,7 +294,6 @@ def generate_config(
         report_stats=None,
         open_private_ports=False,
         listeners=None,
-        database_conf=None,
```
Comment (richvdh) [inline synapse/config/_base.py]: this was never set, outside of tests.

File: docs/postgres.md [hunk 2]
```diff
@@ -105,19 +105,41 @@ of free memory the database host has available.
...
+keepalives_idle: 10
+
+# the number of seconds after which a TCP keepalive message that is not
+# acknowledged by the server should be retransmitted
+keepalives_interval: 10
+
+# the number of TCP keepalives that can be lost before the client's connection
+# to the server is considered dead
+keepalives_count: 3
+```
```
Comment (anoadragon453) [inline docs/postgres.md]: Are these supposed to go at the top-level of the config file or in the `database` section?

Comment (richvdh) [reply docs/postgres.md]: in the `args` section of the `database` section. I was hoping that would be obvious. Suggestions as to how to make it more obvious?

Comment (anoadragon453) [reply docs/postgres.md]: Although the prior paragraph mentions values in `args`, reading through it the connection is still a bit unclear.

I've written the suggestion in another comment.

File: docs/postgres.md [hunk 3]
```diff
@@ -105,19 +105,41 @@ of free memory the database host has available.
...
 
 All key, values in `args` are passed to the `psycopg2.connect(..)`
 function, except keys beginning with `cp_`, which are consumed by the
-twisted adbapi connection pool.
+twisted adbapi connection pool. See the [libpq
+documentation](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-PARAMKEYWORDS)
+for a list of options which can be passed.
+
+You should consider tuning the `keepalives_*` options if there is any danger of
```
Comment (anoadragon453) [inline docs/postgres.md]:
```suggestion
You should consider tuning the `args.keepalives_*` options if there is any danger of
```

**Pull request merged**